week-4/implement/is-prime.test.js

```diff
@@ -1,3 +1,48 @@
 // Given a positive integer num,
 // When the isPrime function is called with num as input,
 // Then it should check if the num is prime
+
+
+function isPrime(num) {
+    if (num <= 1) {
+        return false;
+    }
+    
+    if (num <= 3) {
+        return true;
+    }
+    
+    if (num % 2 === 0 || num % 3 === 0) {
+        return false;
+    }
+    
+    for (let i = 5; i * i <= num; i += 6) {
+        if (num % i === 0 || num % (i + 2) === 0) {
+            return false;
+        }
+    }
+    
+    return true;
+}
```

Comment:
Interesting implementation. It passes the tests, so I can see it works great, but can you explain to me what each line inside this function is doing? I am not wanting this to sya that your implementation is bad or wrong, I just want to see your way of thinking of a solution for this particular problem.